CMakeLists.txt

```diff
@@ -47,43 +53,22 @@ PROJECT(cuda-api-wrappers
 # General C++ build settings
 # ----------------------------
 
+# Here we set a PRIVATE compile option for all targets in this project
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wpedantic" )
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
 
-include_directories( "src/" )
```

Comment:
Why remove this? I mean, it will only apply to this `CMakeLists.txt` 's targets, won't it? Or is this a PRIVATE vs PUBLIC issue?

Reply:
Yes, this is a `PRIVATE` vs `PUBLIC` issue. If you write it as a global statement, it should be a `PRIVATE` include. Instead I added `target_include_directories` in line 92, which also considers different includes for `install` or `build` target. So if I link my own target against `cuda-api-wrappers` the correct include directories are automatically set.